pkg/volume/gce_pd/gce_util.go

```diff
@@ -149,11 +150,59 @@ func (mounter *gceSafeFormatAndMount) Mount(source string, target string, fstype
 	}
 	args = append(args, options...)
 	args = append(args, source, target)
-	glog.V(5).Infof("exec-ing: /usr/share/google/safe_format_and_mount %v", args)
-	cmd := mounter.runner.Command("/usr/share/google/safe_format_and_mount", args...)
+	safeFormatAndMountPath := "/usr/share/google/safe_format_and_mount"
+	glog.V(5).Infof("exec-ing: %s %v", safeFormatAndMountPath, args)
+	cmd := mounter.runner.Command(safeFormatAndMountPath, args...)
 	dataOut, err := cmd.CombinedOutput()
 	if err != nil {
-		glog.V(5).Infof("error running /usr/share/google/safe_format_and_mount\n%s", string(dataOut))
+		glog.V(5).Infof("error running %s\n%s", safeFormatAndMountPath, string(dataOut))
+		// if safe_format_and_mount is not available on this system use linux utilities directly.
+		if _, error := os.Stat(safeFormatAndMountPath); os.IsNotExist(error) {
+			return mounter.formatAndMount(source, target, fstype, options)
+		}
+	}
+	return err
+}
+
+// formatAndMount uses unix utils to format and mount the given disk
+func (mounter *gceSafeFormatAndMount) formatAndMount(source string, target string, fstype string, options []string) error {
+
+	// Add the default options from safe_format_and_mount
+	options = append(options, "defaults")
+
+	// Try to mount the disk. Let 'mount' try to determine the fstype; if it determines the
+	// filesystem type to be "null" then assume the disk is unformatted.
+	err := mounter.Interface.Mount(source, target, "", options)
+	if err != nil && strings.Contains(err.Error(), "unknown filesystem type '(null)'") {
+		// It is very likely that this disk is not formatted. Double check using 'file'
+		var notFormatted bool
+		notFormatted, err = mounter.diskLooksUnformatted(source)
+		if err == nil && notFormatted {
+			// Disk is unformatted so format it.
+			// Use 'ext4' as the default to be compatible with safe_format_and_mount
+			if len(fstype) == 0 {
+				fstype = "ext4"
+			}
+			args := []string{"-E", "lazy_itable_init=0,lazy_journal_init=0", "-F", source}
+			cmd := mounter.runner.Command("mkfs."+fstype, args...)
+			_, err = cmd.CombinedOutput()
+			if err == nil {
+				// the disk has been formatted sucessfully try to mount it again.
+				err = mounter.Interface.Mount(source, target, fstype, options)
+			}
+		}
 	}
 	return err
 }
+
+// diskLooksUnformatted uses 'file' to see if the given disk is unformated
+func (mounter *gceSafeFormatAndMount) diskLooksUnformatted(disk string) (bool, error) {
+	args := []string{"-L", "--special-files", disk}
+	cmd := mounter.runner.Command("file", args...)
+	dataOut, err := cmd.CombinedOutput()
+
+	if err == nil && strings.TrimSpace(string(dataOut)) == disk+": data" {
```

Comment:
https://github.com/GoogleCloudPlatform/compute-image-packages/blob/master/google-startup-scripts/usr/share/google/safe_format_and_mount#L90

The bash script uses *filesystem*, is it a better condition to check?